From d57fcacef4418b2cdbad6a083097fb0c7cad0ca9 Mon Sep 17 00:00:00 2001
From: Rene Engelhard <rene@debian.org>
Date: Sat, 29 Oct 2022 15:23:20 +0200
Subject: [PATCH] require xmlsec1-nss >= 1.2.35

1.2.34 apparently doesn't have xmlSecBase64Decode_ex (needed since
19e998aa98d966cffca98bed9408d5758d91d9be):

[build CXX] xmlsecurity/source/xmlsec/nss/x509certificate_nssimpl.cxx
/home/rene/LibreOffice/git/master/xmlsecurity/source/gpg/SecurityEnvironment.cxx: In member function 'virtual com::sun::star::uno::Reference<com::sun::star::security::XCertificate> SecurityEnvironmentGpg::getCertificate(const rtl::OUString&, const com::sun::star::uno::Sequence<signed char>&)':
/home/rene/LibreOffice/git/master/xmlsecurity/source/gpg/SecurityEnvironment.cxx:166:16: error: 'xmlSecBase64Decode_ex' was not declared in this scope; did you mean 'xmlSecBase64Decode'?
  166 |     int nRet = xmlSecBase64Decode_ex(strKeyId, const_cast<xmlSecByte*>(strKeyId), xmlStrlen(strKeyId), &nWritten);
      |                ^~~~~~~~~~~~~~~~~~~~~
      |                xmlSecBase64Decode
make[3]: *** [/home/rene/LibreOffice/git/master/solenv/gbuild/LinkTarget.mk:337: /home/rene/LibreOffice/git/master/workdir/CxxObject/xmlsecurity/source/gpg/SecurityEnvironment.o] Error 1

and is neither in a grep -r on the header directory

Change-Id: Ie7948ed9b796cc9823e44982824d3c3c61c9a46b
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 8446105fceac..e98a71342a46 100644
--- a/configure.ac
+++ b/configure.ac
@@ -9623,7 +9623,7 @@ libo_CHECK_SYSTEM_MODULE([expat], [EXPAT], [expat])
 dnl ===================================================================
 dnl Check for system xmlsec
 dnl ===================================================================
-libo_CHECK_SYSTEM_MODULE([xmlsec], [XMLSEC], [xmlsec1-nss >= 1.2.28])
+libo_CHECK_SYSTEM_MODULE([xmlsec], [XMLSEC], [xmlsec1-nss >= 1.2.35])
 
 AC_MSG_CHECKING([whether to enable Embedded OpenType support])
 if test "$enable_eot" = "yes"; then
-- 
2.30.2

